testrunner/src/mill/testrunner/TestRunnerUtils.scala

```diff
@@ -135,17 +136,22 @@ import java.io.PrintStream
   private def executeTasks(
       tasks: Seq[Task],
       testReporter: TestReporter,
-      runner: Runner,
       events: ConcurrentLinkedQueue[Event],
       systemOut: PrintStream
-  ): Unit = {
+  ): Boolean = {
+    val taskStatus = new AtomicBoolean(true)
     val taskQueue = tasks.to(mutable.Queue)
     while (taskQueue.nonEmpty) {
       val next = taskQueue.dequeue().execute(
         new EventHandler {
           def handle(event: Event) = {
             testReporter.logStart(event)
             events.add(event)
+            event.status match {
+              case Status.Error => taskStatus.set(false)
+              case Status.Failure => taskStatus.set(false)
+              case _ => taskStatus.compareAndSet(true, true) // consider success as a default
```

Comment:
I think we can leave out this `compareAndSet`; this branch of the `match` being a no-op would be sufficient